groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/completion/CompletionHandler.java

```diff
@@ -53,12 +61,14 @@
 import org.netbeans.modules.groovy.editor.api.lexer.LexUtilities;
 import org.netbeans.modules.groovy.editor.utils.GroovyUtils;
 import org.netbeans.modules.groovy.editor.api.completion.util.CompletionContext;
+import org.netbeans.modules.groovy.editor.java.JavaElementHandle;
 import org.netbeans.modules.groovy.support.api.GroovySettings;
+import org.openide.util.Exceptions;
 import org.openide.util.NbBundle;
 import org.openide.util.Utilities;
 import org.openide.util.WeakListeners;
 
-public class CompletionHandler implements CodeCompletionHandler {
+public class CompletionHandler implements CodeCompletionHandler2 {
```

Comment:
`CodeCompletionHandler2` has been introduced in version of 2.43 of the csl.api. Please update dependency from 2.21 to 2.43.